include/claws/array_ops.hpp

```diff
@@ -169,6 +169,30 @@ namespace claws
     SCALAR_ARRAY_OPERATOR_DEF(<<);
     SCALAR_ARRAY_OPERATOR_DEF(>>);
   }
-
   /// @}
+
+  ///
+  /// \brief struct to allow to speciailize on the retrned array's size
```

Comment:
Could you elaborate on how this can be an advantage (compared to the previous implementation which only used free functions) ?

Reply:
The previous implementation would have required at least two parameters: the size of the array and the functor type. Passing the functor type is cumbersome (requires to have an external lambda and a `decltype`) but is necessary since we can't partially deduce the function's parameters. In the case of a member function however, we can use template parameters from it's enclosing class. This allow on one hand to pass the size to the struct (thus allowing compilation) and deduce the functor's type in the member function (thus allowing to omit it's type) at the cost of two additional brackets when using. Another solution could be using a constructor in a class inheriting from `std::array`, but I prefer the straightforward version.

Reply:
Hum, actually I think partial deduction works as long as it's from "left to right" (like in the example below, the deduced `Functor` type is the rightmost in the template parameter list).

```
template <typename T, size_t size, size_t ...indexes, typename Functor>
std::array<T, size> make_array_impl(std::index_sequence<indexes...>, Functor &&f)
{
  return {((void)indexes, f())...};
}

template <typename T, size_t size, typename Functor>
std::array<T, size> make_array(Functor &&f)
{
  return make_array_impl<T, size>(std::make_index_sequence<size>{}, std::forward<Functor>(f));
}
```

And you could call that like `make_array<int, 3>([](){ return 0; })` if that's OK for you ;)

Reply:
My life will never be the same.